### ISSUE TITLE
Correct way to take screenshots for reporting bugs in a color managed setup?

### ISSUE THREAD

**Issue by CrHasher** [mpv-player/mpv#9826]
- mpv-0.34.0-x86_64
- Windows 11
- https://sourceforge.net/projects/mpv-player-windows/files
- Possible screenshot or video of visual glitches <- This is my problem

This is more of a question than an issue/bug.

How do I correctly take a screen-shot to reflect what appears on screen when reporting bugs?

Relevant options:
icc-profile-auto=yes
screenshot-template="%x\Screens\Screenshot-%F-T%wH.%wM.%wS.%wT-F%{estimated-frame-number}"
screenshot-format=png
screenshot-png-compression=0
screenshot-tag-colorspace=yes
screenshot-high-bit-depth=yes

Calibration info:
I used i1 Display Plus for calibration with DisplayCAL, used the video preset which has a gamma 2.4

What appears on screen when I play a video is totally different than the screenshot I see in a color managed app IrfanView (with color management options turned on). Video is dark the screenshot is bright.

My expectations would be that some info is added to the screen-shot that would allow me to display the captured frame as it was displayed in mpv (regardless of the profile currently used, color management should be able to handle conversion to any profile)

If above can't be achieved than most of the screenshot's in bug reports can't be viewed correctly on another monitor, and irrelevant.

What is the process to take screenshot on my pc and send it to someone else for viewing what I saw on my screen? I can't even do this on my own screen.

Thank you!

**Comment by garamond13** [comment]
Have a look at this issue #9795

**Comment by ValZapod** [comment]
Do printscreen and attach EDID+.icc profile of DISPLAY and some comments on source image...

As for how to report HDR, do JPEG XR screenshots with Nvidia Stream Share API.

**Comment by CrHasher** [comment]
> Do printscreen and attach EDID+.icc profile of DISPLAY and some comments on source image icc profile or video codepoints...
> 
> As for how to report HDR surfaces, do JPEG XR screenshots with Nvidia Stream Share API (overlay that is), simple, that will preserve PQ and scRGB. Of course ideally lossless RGB AVIF tagged as PQ is better.

How do I use the Nvidia Stream Share API?

**Comment by ValZapod** [comment]
> How do I use the Nvidia Stream Share API?

That is just alt+F1 but you need to activate capture on desktop too in overlay settings (Alt+Z). In hdr mode screenshot will be JPEG XR.

>just a 1080p video rec.709/1886 (any video shows same behavior)

If HDR is activated in windows it may still be a good idea.

>brightness is always the same with both gamma 2.2 or gamma 2.4 calibration

Well, yes. With 2.2 gamma it manages 2.4 to 2.2 and when 2.4 it just passes data through.

>calibrate to 2.4 gamma mpv brightens the video

Strange I mean 2.4 gamma calibration darkens all screen and mpv should just preserve it. Maybe if you open PNG image with sRGB (sRGB is 2.2 gamma but with specific linear part in dark region) it will brighten the picture, but that is correct.

>For me it would be ok if mpv assumes that display is calibrated to gamma 2.2 or sRGB, but I want to know if that is the case

That is done by target-trc=srgb or target-trc=gamma2.2

**Comment by CrHasher** [comment]
I will install GeForce Experience for the Alt+F1. Thanks for the idea. Can I expect it to be more accurate than what mpv saves?

I experimented a bit more, it's pretty obvious that with gamma 2.2 the video is brightened and it looses contrast. Calibrated also to sRGB and that looks almost the same as uncalibrated (the monitor is tuned pretty close to ∆E 6500k 120cd regardless of color correcting from software)

My expectation was that sRGB would look less contrasty compared to gamma 2.2, but it's the exact opposite. With sRGB I can barely see strands of hair in dark scenes but they are still visible. With gamma 2.2 I see all strands of hair but contrast is noticeably bad image is washed out.

This is how I calibrated:
![Screenshot 2022-02-05 172431](https://user-images.githubusercontent.com/834518/152657676-d7022ed4-3807-4dcf-b292-f00953147593.png)
![Screenshot 2022-02-05 172442](https://user-images.githubusercontent.com/834518/152657672-580c3245-3c82-4c3c-8e15-337a4a30df33.png)
![Screenshot 2022-02-05 172452](https://user-images.githubusercontent.com/834518/152657680-a06bf598-6fe5-4cdd-b48f-026d2980cc14.png)

**Comment by ValZapod** [comment]
> Can I expect it to be more accurate than what mpv saves?

Yes!!

**Comment by ValZapod** [comment]
> was that sRGB would look less contrasty compared to gamma 2.2

I doubt you can see the difference. Even I have a hard time. Those two are almost the same. sRGB EOTF is almost 2.2 gamma. Just some linear part. When we are talking about OETF it is more visable. You can test with http://cinelex.ru/blog/sravnenie-gammy-v-raznyx-standartax/

Maybe you meant 2.4?

**Comment by garoto** [comment]
Shouldn't `screenshot video` take care of all this by itself (other than the output device color calibration)?

**Comment by CrHasher** [comment]
Never mind this is a duplicate of #8009 & #8082 & #8464 (which I opened silly me)

Using this "fixes" it for HD videos:
`vf=format:gamma=gamma2.2`

What are the implications of these settings?
How exactly should I calibrate to not be forced to use these options?

**Comment by ValZapod** [comment]
> should I calibrate to not be forced to use these options?

It is not just calibration, you should also tell mpv what to target. If you calibrated to bt.1886 you can now play BT.709 movied no problem, but when you play sRGB photo or a video with sRGB transfer upyou should pass --target-trc=bt.1886, because mpv default for bt.1886 for bt.1886 and srgb for srgb, since it tries to not touch by default anything, after all you need to delinearise and the color manage and then linearise again maybe dhanging the transfer, so...

**Comment by CrHasher** [comment]
Isn't  --target-trc ignored when using ICC profile?

What I really need is someone recommending DisplayCal settings that work well with mpv, ideal settings that mpv prefers, I don't really mind switching profile before starting a video.

A table with the most common calibrations and option to use for each tone curve srgb, gamma 2.2,  gamma 2.4, 1886.

I really don't get what the two options do:
`vf=format:gamma=gamma2.2`
And
`--target-trc=whatever`

**Comment by CrHasher** [comment]
Let's get back to the subject about taking screenshots.

The option in mpv pressing 's' works the same as nvidia's Alt+F1, I see no difference between taking a screenshot with mpv or GeForce Experience.

I'm closing this as solved.

**Comment by ValZapod** [comment]
> see no difference between taking a screenshot with mpv or GeForce Experience.

For HDR screenshots the normal app to show the screenshots should be used. HDR +WCG Image viewer from microsoft store.
